crates/nu-protocol/src/signature.rs

```diff
@@ -322,6 +472,39 @@ impl Signature {
         self
     }
 
+    pub fn param(mut self, param: impl Into<Parameter>) -> Self {
+        let param: Parameter = param.into();
+        match param {
+            Parameter::Flag(flag) => {
+                if let Some(s) = flag.short {
+                    debug_assert!(
+                        !self.get_shorts().contains(&s),
+                        "There may be duplicate short flags for '-{s}'"
+                    );
+                }
+
+                let name = flag.long.as_str();
+                debug_assert!(
+                    !self.get_names().contains(&name),
+                    "There may be duplicate name flags for '--{name}'"
+                );
+
+                self.named.push(flag);
+            }
+            Parameter::Required(positional_arg) => {
+                self.required_positional.push(positional_arg);
+            }
+            Parameter::Optional(positional_arg) => {
+                self.optional_positional.push(positional_arg);
+            }
+            Parameter::Rest(positional_arg) => {
+                debug_assert!(self.rest_positional.is_none());
```

Comment:
Just for good measure, could you add a message here as well?

In my opinion we could actually choose to use `assert` for those cases, as for plugin authors etc. that have a worse CI than we have could run into issues with duplicated flags due to conditional compilation. That would be caught by a build even after it slipped through testing.